src/i18n.js

```diff
@@ -63,8 +63,11 @@ const rtlLangs = [ "fa", "ar-SY", "ur" ];
  * @returns {string} the locale that should be displayed
  */
 export function currentLocale() {
-    const potentialLocales = [ localStorage.locale, navigator.language, navigator.language.substring(0, 2), ...navigator.languages ];
-    const availableLocales = potentialLocales.filter(l => languageList[l]);
+    const potentialLocales = [ localStorage.locale, navigator.language, ...navigator.languages ];
+    const availableLocales = potentialLocales
+        .filter(l => !!l)
+        .map(l => l.split("-")[0] in messages ? l.split("-")[0] : l)
+        .filter(l => l in messages);
```

Comment:
Since it is getting more conditions now, is it able to break them down into more traditional programming `for-loop` and `if-else`? Because I am a bit struggling to understand why those bugs is happening.

Also it didn't utilize `languageList` anymore here, make me wondering if it is really ok.

Reply:
I have refactored this to a for-loop.
not checking against `languageList` is because "en" is sort of a magic value amongst all the languages => is only in the `messages` and not in the `languageList`

Reply:
Thanks, I got it now. I am really bad at reading/tracing functional programming style code.🤣